### PR TITLE
feat: improve DX with new dev command and updated contribution guidelines

### PR DESCRIPTION
**Pull request by banjo**
# Description

## TLDR

This
```
npm run build:graphql-docs
npm run build:bruno-query
npm run build:bruno-common

# terminal 1
npm run dev:web

# terminal 2
npm run dev:electron
```

is replaced with this
```
npm run dev
```

## Description

Currently, there are some manual steps included to run the app locally, which includes building all packages locally and then run `next` and `electron` in two separate terminals. This is a bit cumbersome and we also lose the hot reload for the packages that are built beforehand. 

This update adds a new `dev` command that does everything for the user in parallel. Build all packages (which watch mode so it updates automatically when new changes occur), run `next` and run `electron`. All with one command, which makes it much easier for the developer to get up and running.

Here is a gif of me testing the command and just saving a file in the `common` package to show that it hot reloads and includes those updates in the application.

The GIF is too large to include so here it is: https://dsc.cloud/c0ec82/CleanShot-2024-01-30-at-10.11.49.gif

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [ ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

### CODE REVIEW

**Comment by banjo** [comment]
I am not sure why the tests do not work here, I just get this very vague message in `Publish Test Report`:

```
Run dorny/test-reporter@v1
Action was triggered by pull_request: using SHA from head of source branch
Check runs will be created with SHA=74e2c33[14](https://github.com/usebruno/bruno/actions/runs/7871260622/job/21474174252?pr=1477#step:7:15)c2b6bd00f9de76c0de27459f057d625
Listing all files tracked by git
Found 645 files tracked by GitHub
Using test report parser 'java-junit'
Creating test report Test Report
Error: HttpError: Resource not accessible by integration
```

**Comment by helloanoop** [comment]
Closing this PR as this been addressed in https://github.com/usebruno/bruno/pull/2840
